### PR TITLE
docs(openid-connect): session library v4.0.0

### PR DESCRIPTION
**Pull request by samugi**
### Description

Documentation for https://github.com/Kong/kong-ee/pull/4395 / [KAG-234](https://konghq.atlassian.net/browse/KAG-234)

### Checklist 

- [X] Review label added
- [X] PR pointed to correct branch

[KAG-234]: https://konghq.atlassian.net/browse/KAG-234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by kikito — APPROVED** [review]

File: app/_hub/kong-inc/openid-connect/_index.md [hunk 1]
```diff
@@ -920,7 +930,25 @@ params:
...
+        - `Strict`: Cookies will only be sent in a first-party context and not be sent along with requests initiated by third party websites.
+        - `Lax`: Cookies are not sent on normal cross-site subrequests (for example to load images or frames into a third party site), but are sent when a user is navigating to the origin site (for instance, when following a link).
+        - `None`: Cookies will be sent in all contexts, for example in responses to both first-party and cross-origin requests. If `SameSite=None` is set, the cookie Secure attribute must also be set (or the cookie will be blocked).
+        - `Default`: Do not explicitly specify a Same-Site attribute.
     - name: authorization_cookie_httponly
+      maximum_version: "3.1.x"
+      required: false
+      default: true
+      datatype: boolean
+      description: 'Forbids JavaScript from accessing the cookie, for example, through the `Document.cookie` property.'
```
Comment (lena-larionova) [inline app/_hub/kong-inc/openid-connect/_index.md:949]:
```suggestion
      description: | 
        Forbids JavaScript from accessing the cookie, for example, through the `Document.cookie` property.
```

File: app/_hub/kong-inc/openid-connect/_index.md [hunk 2]
```diff
@@ -974,7 +1028,25 @@ params:
...
+    - name: session_cookie_same_site
+      minimum_version: "3.2.x"
+      required: false
+      default: '"Lax"'
+      datatype: string
+      description: |
+        Controls whether a cookie is sent with cross-origin requests, providing some protection against cross-site request forgery attacks:
+        - `Strict`: Cookies will only be sent in a first-party context and aren't sent along with requests initiated by third party websites.
+        - `Lax`: Cookies are not sent on normal cross-site subrequests (for example to load images or frames into a third party site), but are sent when a user is navigating to the origin site (for example, when following a link).
+        - `None`: Cookies will be sent in all contexts, for example in responses to both first party and cross-origin requests. If SameSite=None is set, the cookie Secure attribute must also be set (or the cookie will be blocked)
```
Comment (lena-larionova) [inline app/_hub/kong-inc/openid-connect/_index.md:1040]:
```suggestion
        - `None`: Cookies will be sent in all contexts, for example in responses to both first party and cross-origin requests. If `SameSite=None` is set, the cookie `Secure` attribute must also be set, or the cookie will be blocked.
```

File: app/_hub/kong-inc/openid-connect/_index.md [hunk 3]
```diff
@@ -974,7 +1028,25 @@ params:
...
+        - `Default`: Do not explicitly specify a Same-Site attribute.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/openid-connect/_index.md:1041]:
```suggestion
        - `Default`: Do not explicitly specify a `Same-Site` attribute.
```
Is it Same-Site or SameSite? The previous bullet point has it as SameSite.

File: app/_hub/kong-inc/openid-connect/_index.md [hunk 4]
```diff
@@ -974,7 +1028,25 @@ params:
...
     - name: session_cookie_httponly
+      maximum_version: "3.1.x"
+      required: false
+      default: true
+      datatype: boolean
+      description: 'Forbids JavaScript from accessing the cookie, for example, through the `Document.cookie` property.'
```
Comment (lena-larionova) [inline app/_hub/kong-inc/openid-connect/_index.md:1047]:
```suggestion
      description: |
        Forbids JavaScript from accessing the cookie, for example, through the `Document.cookie` property.
```

File: app/_hub/kong-inc/openid-connect/_index.md [hunk 5]
```diff
@@ -1025,13 +1099,96 @@ params:
...
       description: |
         The session strategy:
         - `none`: no compression
         - `zlib`: use zlib to compress cookie data
+    - name: session_audience
+      minimum_version: "3.2.x"
+      required: false
+      default: '"default"'
+      datatype: string
+      description: The session audience, for example "my-application"
```
Comment (lena-larionova) [inline app/_hub/kong-inc/openid-connect/_index.md:1115]:
```suggestion
      description: The session audience, which is the intended target application. For example `"my-application"`.
```

File: app/_hub/kong-inc/openid-connect/_index.md [hunk 6]
```diff
@@ -1025,13 +1099,96 @@ params:
...
+    - name: session_remember
+      minimum_version: "3.2.x"
+      required: false
+      default: false
+      datatype: boolean
+      description: Enables or disables persistent sessions
```
Comment (lena-larionova) [inline app/_hub/kong-inc/openid-connect/_index.md:1121]:
```suggestion
      description: Enables or disables persistent sessions.
```

File: app/_hub/kong-inc/openid-connect/_index.md [hunk 7]
```diff
@@ -1025,13 +1099,96 @@ params:
...
+    - name: session_remember_cookie_name
+      minimum_version: "3.2.x"
+      required: false
+      default: '"remember"'
+      datatype: string
+      description: Persistent session cookie name
```
Comment (lena-larionova) [inline app/_hub/kong-inc/openid-connect/_index.md:1127]:
```suggestion
      description: Persistent session cookie name. Use with the `remember` configuration parameter.
```

File: app/_hub/kong-inc/openid-connect/_index.md [hunk 8]
```diff
@@ -1025,13 +1099,96 @@ params:
...
+    - name: session_remember_rolling_timeout
+      minimum_version: "3.2.x"
+      required: false
+      default: 604800
+      datatype: integer
+      description: Persistent session rolling timeout in seconds.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/openid-connect/_index.md:1133]:
```suggestion
      description: The persistent session rolling timeout window, in seconds.
```

File: app/_hub/kong-inc/openid-connect/_index.md [hunk 9]
```diff
@@ -1025,13 +1099,96 @@ params:
...
+    - name: session_remember_absolute_timeout
+      minimum_version: "3.2.x"
+      required: false
+      default: 2592000
+      datatype: integer
+      description: Persistent session absolute timeout in seconds.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/openid-connect/_index.md:1139]:
```suggestion
      description: The persistent session absolute timeout limit, in seconds.
```

File: app/_hub/kong-inc/openid-connect/_index.md [hunk 10]
```diff
@@ -1025,13 +1099,96 @@ params:
...
+    - name: session_request_headers
+      minimum_version: "3.2.x"
+      required: false
+      default: null
+      datatype: array of string elements
+      description: |
+        List of information to include (as headers) in the request to upstream. 
+        Accepted values are: `id`, `audience`, `subject`, `timeout`, `idling-timeout`, `rolling-timeout`,
+        `absolute-timeout`.
+        For example, { "id", "timeout" } will set both `Session-Id` and `Session-Timeout` in the request headers.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/openid-connect/_index.md:1149]:
```suggestion
        List of information to include, as headers, in the request to the upstream. 
        Accepted values are: `id`, `audience`, `subject`, `timeout`, `idling-timeout`, `rolling-timeout`, and
        `absolute-timeout`.
        For example, `{ "id", "timeout" }` sets both `Session-Id` and `Session-Timeout` in the request headers.
```

File: app/_hub/kong-inc/openid-connect/_index.md [hunk 11]
```diff
@@ -1025,13 +1099,96 @@ params:
...
+    - name: session_response_headers
+      minimum_version: "3.2.x"
+      required: false
+      default: null
+      datatype: array of string elements
+      description: |
+        List of information to include (as headers) in the response to downstream. 
+        Accepted values are: `id`, `audience`, `subject`, `timeout`, `idling-timeout`, `rolling-timeout`,
+        `absolute-timeout`.
+        For example, { "id", "timeout" } will inject both `Session-Id` and `Session-Timeout` in the response headers.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/openid-connect/_index.md:1159]:
```suggestion
        List of information to include, as headers, in the response to the downstream. 
        Accepted values are: `id`, `audience`, `subject`, `timeout`, `idling-timeout`, `rolling-timeout`, and
        `absolute-timeout`.
        For example: `{ "id", "timeout" }` injects both `Session-Id` and `Session-Timeout` in the response headers.
```

File: app/_hub/kong-inc/openid-connect/_index.md [hunk 12]
```diff
@@ -1025,13 +1099,96 @@ params:
...
+    - name: session_store_metadata
+      minimum_version: "3.2.x"
+      required: false
+      default: false
+      datatype: boolean
+      description: | 
+        Configures whether or not session metadata should be stored.
+        This includes information about the active sessions for a specific audience
```
Comment (lena-larionova) [inline app/_hub/kong-inc/openid-connect/_index.md:1167]:
```suggestion
        This metadata includes information about the active sessions for a specific audience
```

**Review by lena-larionova — CHANGES_REQUESTED** [review]
Mostly the same edits as for the Session plugin.
You'll want to add any new parameter names to the plugin ignore list to avoid getting the linter comments again: https://github.com/Kong/docs.konghq.com/blob/docs/openid-connect-session-lib-v4/.github/styles/kongplugins/plugin-ignore.txt

File: app/_hub/kong-inc/openid-connect/_index.md (hunk 3 above)
Comment (samugi) [reply app/_hub/kong-inc/openid-connect/_index.md]: fixed it, thanks

**Review by lena-larionova — APPROVED** [review]
LGTM, thanks for making these changes. 

Took a quick look through the docs, the deprecated params aren't used anywhere, so no other edits needed.

**Pull request merged**